group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -598,15 +600,16 @@ public List<DescribeGroupsResponseData.DescribedGroup> describeGroups(
      */
     ConsumerGroup getOrMaybeCreateConsumerGroup(
         String groupId,
-        boolean createIfNotExists
+        boolean createIfNotExists,
+        List<Record> records
     ) throws GroupIdNotFoundException {
         Group group = groups.get(groupId);
 
         if (group == null && !createIfNotExists) {
             throw new GroupIdNotFoundException(String.format("Consumer group %s not found.", groupId));
         }
 
-        if (group == null) {
+        if (group == null || maybeDeleteEmptyClassicGroup(group, records)) {
```

Comment:
Should we only consider deleting the previous empty group only if `createIfNotExists` is `true`? It is perhaps not necessary. What do you think?

Reply:
Oh thanks for the catch. Yeah it's necessary when the group is a empty classic group